pkg/gateway/operations/base.go

```diff
@@ -150,7 +150,7 @@ func (o *Operation) EncodeError(w http.ResponseWriter, req *http.Request, e erro
 
 func generateHostID() string {
 	const generatedHostIDLength = 8
-	return auth.HexStringGenerator(generatedHostIDLength)
+	return compat.HexStringGenerator(generatedHostIDLength)
```

Comment:
```suggestion
	return keys.HexStringGenerator(generatedHostIDLength)
```
?

Reply:
good catch - I think less inner-packages less issues

Reply:
Agreed.  But exporting all of `auth` drags in too much.  I just want to generate AWS credentials somewhere else.